03week/towersOfHanoi.js

```diff
@@ -13,34 +21,86 @@ let stacks = {
   c: []
 };
 
+let moveCount = 0;
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  // You can only move the top number (pop method) of any stack, and you can only move one number at a time, and you move the number the another stack with the push method.
+  // const userObj = stack[].pop();
+  if (isValid(startStack, endStack) && isLegal(startStack, endStack)) {
+    stacks[endStack].push(stacks[startStack].pop());
+    moveCount++;
+  } else {
+    console.log("Please try again!");
+    return false;
+  }
 }
 
-function isLegal() {
-  // Your code here
+function isValid(startStack, endStack) {
+  if ((startStack === "a" || startStack === "b" || startStack === "c") && (endStack === "a" || endStack === "b" || endStack === "c")) {
```

Comment:
This could be simplified, you could have an array with available letters and see if any are in each stack.